python/lsst/meas/algorithms/objectSizeStarSelector.py

```diff
@@ -450,37 +469,12 @@ def selectStars(self, exposure, sourceCat, matches=None):
 
                     ds9.dot("+", source.getX() - mi.getX0(),
                             source.getY() - mi.getY0(), frame=frame, ctype=ctype)
-        #
-        # Time to use that stellar classification to generate psfCandidateList
-        #
-        with ds9.Buffering():
-            psfCandidateList = []
-            for isStellar, source in zip(stellar, [s for g, s in zip(good, sourceCat) if g]):
-                if not isStellar:
-                    continue
 
-                try:
-                    psfCandidate = algorithmsLib.makePsfCandidate(source, exposure)
-                    # The setXXX methods are class static, but it's convenient to call them on
-                    # an instance as we don't know Exposure's pixel type
-                    # (and hence psfCandidate's exact type)
-                    if psfCandidate.getWidth() == 0:
-                        psfCandidate.setBorderWidth(self._borderWidth)
-                        psfCandidate.setWidth(self._kernelSize + 2*self._borderWidth)
-                        psfCandidate.setHeight(self._kernelSize + 2*self._borderWidth)
-
-                    im = psfCandidate.getMaskedImage().getImage()
-                    vmax = afwMath.makeStatistics(im, afwMath.MAX).getValue()
-                    if not numpy.isfinite(vmax):
-                        continue
-                    psfCandidateList.append(psfCandidate)
-
-                    if display and displayExposure:
-                        ds9.dot("o", source.getX() - mi.getX0(), source.getY() - mi.getY0(),
-                                size=4, frame=frame, ctype=ds9.CYAN)
-                except Exception as err:
-                    logger.logdebug("Failed to make a psfCandidate from source %d: %s" % (source.getId(), err))
-
-        return psfCandidateList
-
-starSelectorRegistry.register("objectSize", ObjectSizeStarSelector)
+        starCat = SourceCatalog(sourceCat.schema)
+        for isStellar, source in zip(stellar, [s for g, s in zip(good, sourceCat) if g]):
```

Comment:
In my opinion the code is confusing and needs a cleanup pass. However, I will break this one line into two (make the list, then use it). It is still not obvious to me why `stellar` and `good` have the same length (e.g. from reading the code above).